File: src/test/test_compile/input/test0.pas

program fib(input, output);
var
  a, i: integer;
function fib(y: integer): integer;
begin
if y = 0 then
    fib := 0
  else
  begin
    if y = 1 then
      fib := 1
    else
      fib := fib(y - 1) + fib(y - 2);
  end;
end;

begin
  for i:= 1 to 10 do
    write(fib(i),' ');
end.

File: test/test_compile_1/input/c_code/test0.c
#include <stdio.h>

int a,i;

int fib(int y)
{
	int _fib;
	if (y==0)
		_fib = 0;
	else
	{
		if (y==1)
			_fib = 1;
		else
			_fib = fib(y-1)+fib(y-2);
	}
	return _fib;
}

int main()
{
	for(i = 1; i <= 10; i ++)
		printf("%d%c", fib(i), ' ');
	return 0;
}